download_and_install/compiling_from_source.md

```diff
@@ -0,0 +1,242 @@
+
+- [avahi](https://www.avahi.org) if you want to use network discovery.
+- [microhttpd](https://www.gnu.org/software/libmicrohttpd/) (if you want the web UI).
+- autoconf
+- automake
+- cppunit
+- ncurses
+- lex (or flex)
+- libftdi
+- liblo
+- libtool
+- libusb
+- pkg-config
+- [Protobuf](https://developers.google.com/protocol-buffers) (version 3.6 or later)
+- uuid or ossp uuid
+- yacc (or bison)
+- zlib
+
+The Python library also requires
+
+- Numpy
+- Protobuf for Python
+
+Debian/Ubuntu
+-------------
+
+Debian/Ubuntu users can install all dependencies with:
+
+    sudo apt-get install \
+        autoconf \
+        automake \
+        bison \
+        flex \
+        g++ \
+        libavahi-client-dev \
+        libcppunit-dev \
+        libftdi-dev \
+        liblo-dev \
+        libmicrohttpd-dev \
+        libncurses-dev \
+        libprotobuf-dev \
+        libprotoc-dev \
+        libtool \
+        libusb-1.0.0-dev \
+        make \
+        protobuf-compiler \
+        pkg-config \
+        uuid-dev \
+        zlib1g-dev
+
+CentOS/RHEL/Fedora
+------------------
+
+Distributions using RPM packages can install all dependencies with:
+
+    sudo dnf install \
+        autoconf \
+        automake \
+        avahi-devel \
+        bison \
+        cppunit-devel \
+        flex \
+        gcc-c++ \
+        libftdi-devel \
+        liblo-devel \
+        libmicrohttpd-devel \
+        libtool \
+        libusbx-devel \
+        make \
+        ncurses-devel \
+        protobuf-lite-devel \
+        uuid-devel \
+        zlib-devel
+
+macOS
+-----
+
+Dependencies are available from [MacPorts](https://www.macports.org/):
+
+    sudo port install \
+        autoconf \
+        automake \
+        bison \
+        cppunit \
+        flex \
+        libftdi1 \
+        liblo \
+        libmicrohttpd \
+        libtool \
+        libusb \
+        libuuid \
+        ncurses \
+        pkgconfig \
+        protobuf3-cpp \
+        zlib
+
+or [Homebrew](https://brew.sh/):
+
+    brew install \
+        autoconf \
+        automake \
+        bison \
+        cppunit \
+        flex \
+        libftdi \
+        liblo \
+        libmicrohttpd \
+        libtool \
+        libusb \
+        libuuid \
+        ncurses \
+        pkg-config \
+        protobuf \
+        zlib
+
+Python dependencies
+-------------------
+
+If you wish to build the Python library, additional libraries are required:
+
+    pip3 install --user \
+        numpy \
+        protobuf
+
+If the pip3 command does not exist, install python3-pip from your package manager and try again.
+
+Build
+=====
+
+If this is the first time compiling with this download, generate the build system:
+
+    autoreconf -i
+
+OLA uses autotools for building. Run
+
+     ./configure --help
+
+to see all options. The most popular option is `--enable-python-libs` to build the Python client module. If you want to
+use the RDM responder tests, add `--enable-rdm-tests`.
```

Comment:
It might be worth mentioning `--disable-all-plugins --enable-foo` which I recommend if you only need the foo plugin and you're compiling on something slow like an older Pi.

Reply:
Once I've figured out the Python shenanigans, I might rework this section.  Right now, it's copied more-or-less verbatim from the old text.  I am wary of getting too detailed; don't want to risk letting the docs and code get badly out of sync again.

Reply:
It's unlikely the code will change in this sense, we added the `--disable-all-plugins` to configure to make things like this easier. The `--enable-foo` would be fetched from `./configure --help` depending on which plugin you actually wanted to use.

Backwards compatibility wise, if we added an e.g. --only-foo (which I doubt we would, as realistically you probably want `--only-e131 --only-artnet` which name wise doesn't really make sense, we'd keep the old options for compatibility too).